src/components/CategorySidebar/Category.component.jsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { activeCategory, deactiveCategory } from '../../actions/categories';
+import {
+  removeProductsByCategory,
+  setProductsByCategory,
+} from '../../actions/products';
+import {
+  finishLoadingCategories,
+  startLoadingCategories,
+} from '../../actions/ui';
+import { CategoryActive, CategoryNotActive } from './CategorySidebar.style';
+import PropTypes from 'prop-types';
+
+const Category = ({ id, category, isActive }) => {
+  const dispatch = useDispatch();
+  const handleAdd = () => {
+    dispatch(startLoadingCategories());
+    setTimeout(() => {
+      dispatch(finishLoadingCategories());
+      dispatch(activeCategory(id));
+      dispatch(setProductsByCategory(id));
+    }, 2000);
+  };
+
+  const handleRemove = () => {
+    dispatch(deactiveCategory(id));
```

Comment:
This not should be handled by the redux state, it could be only a `hook` to do it.